Dependencies/xr/Include/XR.h

```diff
@@ -120,6 +112,61 @@ namespace xr
         Identifier ID{};
     };
 
+    struct FieldOfView
+    {
+        float AngleLeft;
+        float AngleRight;
+        float AngleUp;
+        float AngleDown;
+    };
+
+    enum class DetectionBoundaryType
+    {
+        Box,
+        Frustum,
+        Sphere
+    };
+
+    struct Frustum
+    {
+        Pose Pose{};
+        FieldOfView FOV{};
+        float FarDistance;
+    };
+
+    struct DetectionBoundary
+    {
+        DetectionBoundaryType Type{ DetectionBoundaryType::Sphere };
+        std::variant<float, Frustum, Vector3f> Data{ 5.f };
+    };
+
+    struct GeometryDetectorOptions
+    {
+        xr::DetectionBoundary DetectionBoundary{};
+        double UpdateInterval{10};
+    };
+
+    enum class SceneObjectType
+    {
+        Unknown,
+        Background,
+        Wall,
+        Floor,
+        Ceiling,
+        Platform,
+        Undefined
```

Comment:
nit: Is `Undefined` meant to silently communicate an error in recognition? Would it suffice to just have `Unknown` as the default and check if `ID == INVALID_ID` to see if the object has been initialized?

Reply:
Unknown is a type specific to OpenXR. ARKit has its own enumeration. I think we should relook this when consolidating other platform's scene object/compound mesh types. I had been using Unknown with the expectation that a platform attempted enumerating the object but did not know what it was. vs Undefined where the platform did not attempt any enumeration/does not support enumeration. But, I see how this can be confusing/improved.